### ISSUE TITLE
wrong Lint/AssignmentInCondition

### ISSUE THREAD

**Issue by daixiang0** [bbatsov/rubocop#4670]
```
def p
  111
end
if b
  do_something
elsif a = p  <===
  do_something(a)
end
```
It trigger Lint/AssignmentInCondition and said:
```
Assignment in condition - you probably meant to use ==.
```
Clearly, use `==` is wrong.

**Comment by pocke** [comment]
> Clearly, use == is wrong.

I don't think so.
Why do you think this is clearly wrong?

**Comment by daixiang0** [comment]
Since i want to use a new val not compare the val, the cop output is not right.

**Comment by daixiang0** [comment]
Also, if use `elsif(a = p)`, this cop will disappear. I mean the cop output may mislead.

**Comment by bquorning** [comment]
The unclear error message has been fixed on master branch by #4605, but hasn’t been released yet.

> Also, if use `elsif(a = p)`, this cop will disappear.

That sounds like a bug in `Lint/AssignmentInCondition`
